.github/workflows/golangci-lint.yml

```diff
@@ -16,3 +18,23 @@ jobs:
         uses: golangci/golangci-lint-action@v2
         with:
           version: v1.42
+
+  release:
```

Comment:
This new functionality should probably be it's own workflow (release.yaml) that only runs when you push a tag not attached to the current one (linter), which as @petewall mentioned, it runs in every push to main and pull request.

Reply:
I see

BTW in my tests I saw this release running in all commits, which we do not want. But it is true that only when you tag main and push that tag is when you get a release done.

will try to make those chages happen, but will test them in my private repo first.

Reply:
I have done this in my repo test, seems to work fine.

Reply:
The `release.yaml` is now separated in this PR.